homeassistant/components/freebox/__init__.py

```diff
@@ -29,62 +28,82 @@
 
 
 async def async_setup(hass, config):
-    """Set up the Freebox component."""
+    """Set up the Freebox component from legacy config file."""
     conf = config.get(DOMAIN)
 
     async def discovery_dispatch(service, discovery_info):
         if conf is None:
             host = discovery_info.get("properties", {}).get("api_domain")
             port = discovery_info.get("properties", {}).get("https_port")
             _LOGGER.info("Discovered Freebox server: %s:%s", host, port)
-            await async_setup_freebox(hass, config, host, port)
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN,
+                    context={"source": SOURCE_DISCOVERY},
+                    data={CONF_HOST: host, CONF_PORT: port},
+                )
+            )
 
     discovery.async_listen(hass, SERVICE_FREEBOX, discovery_dispatch)
```

Comment:
There's no ssdp or other standard discovery service we can use?

Reply:
As in the freebox docs we can use mDNS or http/https request to http(s)://mafreebox.free.fr which is always resolved internally in the LAN.
For mDNS : The Freebox broadcasts the “_fbx-api._tcp” service.
Then from these methods we can extract the api_domain and port. 
Thing is that the aiofreepybox used here may evolve to use only http://mafreebox.free.fr that is obviously not secure but really local polling. Using it with https would end to a certificate error (NET::ERR_CERT_COMMON_NAME_INVALID)

Reply:
Haven't looked at it and can be done in an other PR I think.